src/c2goto/library/libm/exp.c

```diff
@@ -0,0 +1,135 @@
+
+#include <math.h>
+
+static double expm1_taylor(double x)
+{
+  /* Compute truncated Taylor series for e^x - 1 around 0:
+   * x + x^2/2! + x^3/3! + x^4/4! + ... + x^n/n! */
+  double acc = x;
+  double smd = x;
+  acc += (smd *= x / 2);
+  acc += (smd *= x / 3);
+  acc += (smd *= x / 4);
+  acc += (smd *= x / 5);
+  acc += (smd *= x / 6);
```

Comment:
The `--goto-unwind` should be smart enough if you convert this into a for loop with a constant bound.

Reply:
But this would require the users always to invoke the `--goto-unwind` :-)

Reply:
> But this would require the users always to invoke the --goto-unwind :-)

If we set it to default at `c2goto` that would not be a requirement.

Reply:
does --goto-unwind unwind all loops or is it only the ones with known loop bounds?

Reply:
> only the ones with known loop bounds

Only loops of the like  `for(int i =0; i< N; i++)` where `i` is read-only at the loop body, and `N` is a constant (it needs to be a value).

Reply:
Even if it works with c2goto, there are 2 reasons I wouldn't recommend doing that (here):
1. c2goto is optional, it would be non-intuitive to have the precompiled libs behave different than the non-precompiled ones
2. the *_taylor() functions are coded suboptimally wrt. precision: it's in almost all cases better to use some kind of Horner scheme to evaluate polynomials; I haven't done that in order to be able to easily tweak the order of the polynomials as I was aiming for good verification times

The second point does not apply to pow_by_squaring() and there an unrolled loop would be nice. But the first point still bothers me.